src/material-experimental/mdc-slider/slider-adapter.ts

```diff
@@ -8,131 +8,146 @@
   }
-  addClass(className: string): void {
-    throw Error('Method not implemented.');
+  addClass = (className: string): void => {
+    this._delegate._hostElement.classList.add(className);
   }
-  removeClass(className: string): void {
-    throw Error('Method not implemented.');
+  removeClass = (className: string): void => {
+    this._delegate._hostElement.classList.remove(className);
   }
-  getAttribute(attribute: string): string | null {
-    throw Error('Method not implemented.');
+  getAttribute = (attribute: string): string | null => {
+    return this._delegate._hostElement.getAttribute(attribute);
   }
-  addThumbClass(className: string, thumb: Thumb): void {
-    throw Error('Method not implemented.');
+  addThumbClass = (className: string, thumb: Thumb): void => {
+    this._delegate._getThumbElement(thumb).classList.add(className);
   }
-  removeThumbClass(className: string, thumb: Thumb): void {
-    throw Error('Method not implemented.');
+  removeThumbClass = (className: string, thumb: Thumb): void => {
+    this._delegate._getThumbElement(thumb).classList.remove(className);
   }
-  getInputValue(thumb: Thumb): string {
-    throw Error('Method not implemented.');
+  getInputValue = (thumb: Thumb): string => {
+    return this._delegate._getInputElement(thumb).value;
   }
-  setInputValue(value: string, thumb: Thumb): void {
-    throw Error('Method not implemented.');
+  setInputValue = (value: string, thumb: Thumb): void => {
+    this._delegate._getInputElement(thumb).value = value;
   }
-  getInputAttribute(attribute: string, thumb: Thumb): string | null {
-    throw Error('Method not implemented.');
+  getInputAttribute = (attribute: string, thumb: Thumb): string | null => {
+    return this._delegate._getInputElement(thumb).getAttribute(attribute);
   }
-  setInputAttribute(attribute: string, value: string, thumb: Thumb): void {
-    throw Error('Method not implemented.');
+  setInputAttribute = (attribute: string, value: string, thumb: Thumb): void => {
+    this._delegate._getInputElement(thumb).setAttribute(attribute, value);
   }
-  removeInputAttribute(attribute: string, thumb: Thumb): void {
-    throw Error('Method not implemented.');
+  removeInputAttribute = (attribute: string, thumb: Thumb): void => {
+    this._delegate._getInputElement(thumb).removeAttribute(attribute);
   }
-  focusInput(thumb: Thumb): void {
-    throw Error('Method not implemented.');
+  focusInput = (thumb: Thumb): void => {
+    this._delegate._getInputElement(thumb).focus();
   }
-  isInputFocused(thumb: Thumb): boolean {
-    throw Error('Method not implemented.');
+  isInputFocused = (thumb: Thumb): boolean => {
+    return this._delegate._getInput(thumb)._isFocused();
   }
-  getThumbKnobWidth(thumb: Thumb): number {
-    throw Error('Method not implemented.');
+  getThumbKnobWidth = (thumb: Thumb): number => {
+    return this._delegate._getKnobElement(thumb).getBoundingClientRect().width;
```

Comment:
This could be problematic for SSR depending on when the method is called. Just make sure to add the MDC-slider to the kitchen sink SSR app at some point (can be a future PR)

Reply:
I would add a TODO here to mark this